aoservicelisting.civix.php

```diff
@@ -175,34 +195,41 @@ public static function endABARelationship($values, $rowNumber, $orgId) {
 
   public static function findDupes($cid, $orgId, &$individualParams, $rel = EMPLOYER_CONTACT_REL, $isOrgOptional = FALSE) {
     if (!empty($cid)) {
-      $currentDetails = civicrm_api3('Contact', 'getsingle', ['id' => $cid]);
-      if ($currentDetails['first_name'] != $individualParams['first_name'] || $currentDetails['last_name'] != $individualParams['last_name']) {
-        $params = ['contact_id_a' => $cid, 'contact_id_b' => $orgId, 'is_active' => 1];
-        $relationships = civicrm_api3('Relationship', 'get', $params);
-        if (!empty($relationships['values'])) {
-          // End Date all relationships as they have either overwritten the data or not.
-          foreach ($relationships['values'] as $relationship) {
-            civicrm_api3('Relationship', 'create', ['id' => $relationship['id'], 'is_active' => 0, 'relationship_type_id' => $relationship['relationship_type_id'], 'end_date' => date('Y-m-d')]);
-          }
-        }
-      } else {
-        $individualParams['contact_id'] = $cid;
+      // Check to see if this contact is a primary contact.
+      if (E::checkPrimaryContact($cid)) {
+        // Prevent changing first name and last name for primary contacts.
+        $primaryContact = civicrm_api3('Contact', 'getsingle', [
+          'return' => ['first_name', 'last_name'],
+          'id' => $cid,
+        ]);
+        $individualParams['first_name'] = $primaryContact['first_name'];
```

Comment:
@Edzelopez this feels a little problematic. Here suppose someone has replaced the regulated staff details of the primary contact with person B. This seems like it will mean that person B will be ignored completely i.e. no new record would be created.

Reply:
Would it not make sense to retain the original lines but in L183 put in an extra `if (E::checkPrimaryContact($cid) { return; }`